org-code-javabuilder/protocol/src/main/java/org/code/protocol/MetricClientManager.java

```diff
@@ -0,0 +1,31 @@
+package org.code.protocol;
+
+public class MetricClientManager {
+  private static MetricClientManager clientManagerInstance;
+
+  private MetricClient metricClient;
+
+  private MetricClientManager(MetricClient metricClient) {
+    this.metricClient = metricClient;
+  }
+
+  public static void create(MetricClient metricClient) {
+    MetricClientManager.clientManagerInstance = new MetricClientManager(metricClient);
+  }
+
+  public static MetricClientManager getInstance() {
+    if (MetricClientManager.clientManagerInstance == null) {
+      throw new InternalServerRuntimeException(InternalExceptionKey.INTERNAL_EXCEPTION);
```

Comment:
nit: should we include an Exception cause here so we know what happened? ex. `new Exception("MetricsClient not initialized")`

Reply:
done!